phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/PhoenixConfigurationUtil.java

```diff
@@ -866,4 +872,18 @@ public static void setTenantId(Configuration configuration, String tenantId){
         configuration.set(MAPREDUCE_TENANT_ID, tenantId);
     }
 
+    public static void setMRSnapshotManagedInternally(Configuration configuration, Boolean isSnapshotRestoreManagedInternally) {
```

Comment:
I think should be setMRSnapshotManagedExternally. That keeps it closer to the HBase solution this is modeled after (HBASE-18806) where true means that the external job has already restored the snapshot and will clean it up.